modules/commons/src/main/java/org/apache/synapse/commons/json/jsonprocessor/parser/JsonProcessor.java

```diff
@@ -0,0 +1,141 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.synapse.commons.json.jsonprocessor.parser;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.commons.json.jsonprocessor.constants.ValidatorConstants;
+import org.apache.synapse.commons.json.jsonprocessor.exceptions.ParserException;
+import org.apache.synapse.commons.json.jsonprocessor.exceptions.ValidatorException;
+import org.apache.synapse.commons.json.jsonprocessor.utils.GSONDataTypeConverter;
+import org.apache.synapse.commons.json.jsonprocessor.validators.ArrayValidator;
+import org.apache.synapse.commons.json.jsonprocessor.validators.BooleanValidator;
+import org.apache.synapse.commons.json.jsonprocessor.validators.NullValidator;
+import org.apache.synapse.commons.json.jsonprocessor.validators.NumericValidator;
+import org.apache.synapse.commons.json.jsonprocessor.validators.ObjectValidator;
+import org.apache.synapse.commons.json.jsonprocessor.validators.StringValidator;
+
+/**
+ * This class will parse a given JSON input according to a given schema.
+ * Supported inout formats - String and Gson JsonObject
+ */
+public class JsonProcessor {
+
+    // Use without instantiating
+    private JsonProcessor() {
+    }
+
+    // Logger instance
+    private static Log logger = LogFactory.getLog(JsonProcessor.class.getName());
+
+    // JSON parser instance
+    private static JsonParser parser = new JsonParser();
+
+    /**
+     * This method parse a given JSON string according to the given schema. Both as string.
+     *
+     * @param inputString input String.
+     * @param inputSchema input Schema.
+     * @return corrected String.
+     * @throws ValidatorException Exception occurs in validation process.
+     * @throws ParserException    Exception occurs in data type parsing.
+     */
+    public static String parseJson(String inputString, String inputSchema) throws ValidatorException, ParserException {
+        if (inputString != null && !inputString.isEmpty() && inputSchema != null && !inputSchema.isEmpty()) {
+            JsonObject schemaObject;
+            JsonElement schema = parser.parse(inputSchema);
+            if (schema.isJsonObject()) {
+                schemaObject = schema.getAsJsonObject();
+            } else if (schema.isJsonPrimitive()) {
+                // if schema is primitive it should be a boolean
+                boolean valid = schema.getAsBoolean();
+                if (valid) {
+                    return inputString;
+                } else {
+                    ValidatorException exception = new ValidatorException("JSON schema is not valid for all elements");
+                    logger.error("JSON schema is false, so all validations will fail", exception);
```

Comment:
why do we log and throw here? shall we append this error message to validator exception.

Reply:
fixed